docs/core/custom-models.md

```diff
@@ -0,0 +1,538 @@
+    }
+  ]
+}
+```
+
+means that road segments with `road_class==MOTORWAY` and `road_environment==TUNNEL` get priority `0.5*0.1=0.05` and
+those with `road_class==SECONDARY` and no TUNNEL, get priority `0.9` and so on.
+
+Edges with lower priority values will be less likely part of the optimal route calculated by GraphHopper, higher values
+mean that these road segments shall be preferred. If you do not want to state which road segments shall be avoided, but
+rather which ones shall be preferred, you need to **decrease** the priority of others:
+
+```json
+{
+  "priority": [
+    {
+      "if": "road_class != CYCLEWAY",
+      "multiply_by": 0.8
+    }
+  ]
+}
+```
+
+means decreasing the priority for all road_classes *except* cycleways.
+
+Just like we saw for `speed` you can also adjust the priority for road segments in a certain area. It works exactly the
+same way:
+
+```json
+{
+  "priority": [
+    {
+      "if": "in_custom1",
+      "multiply_by": 0.7
+    }
+  ]
+}
+```
+
+To block an entire area set the priority value to `0`. You can even set the priority only for certain roads in an area
+like this:
+
+```json
+{
+  "priority": [
+    {
+      "if": "road_class == MOTORWAY && in_custom1",
+      "multiply_by": 0.1
+    }
+  ]
+}
+```
+
+Some other useful encoded values to restrict access to certain roads depending on your vehicle dimensions are the
+following:
+
+```json
+{
+  "priority": [
+    {
+      "if": "max_width < 2.5",
+      "multiply_by": 0
+    },
+    {
+      "if": "max_length < 10",
+      "multiply_by": 0
+    },
+    {
+      "if": "max_weight < 3.5",
+      "multiply_by": 0
+    }
+  ]
+}
+```
+
+which means that the priority for all road segments that allow a maximum vehicle width of `2.5m`, a maximum vehicle
+length of `10m` or a maximum vehicle weight of `3.5tons`, or less, is zero, i.e. these "narrow" road segments are
+blocked.
+
+### Customizing `distance_influence`
+
+We already explained the meaning of `distance_influence` in one of the previous sections. To specify its value simply
+use the `distance_influence` property of the custom value like this:
+
+```json
+{
+  "distance_influence": 100
+}
+``` 
+
+If you do not use this property, GraphHopper will use the default value which is `70`.
+
+## How to use Custom Models
+
+### via HTTP
+
+Using custom models is simple. You just need to add the `custom_model` property to your request JSON. This only works
+for certain profiles, however, and so far only POST /route (but not GET /route or /isochrone, /spt or /map-matching)
+are supported.
+
+### via GraphHopper maps
+
+GraphHopper maps offers an interactive text editor that can be used to comfortably enter custom models. You can open it
+by pressing the 'custom' button. It will check the syntax of your custom model and mark errors in red. You can press
+Ctrl+Space or Alt+Enter to retrieve auto-complete suggestions. Pressing Ctrl+Enter will send a routing request for the
+custom model you entered.
```

Comment:
I would even move this section to the top of the description?

What I find also important is the show a complete example e.g. "The POST body for a query in Berlin looks like:"

```json
{
  "points": [
    [ 13.31543, 52.509535 ],
    [ 13.29779, 52.512434 ]
  ],
  "custom_model": {
    "priority": [
      {
        "if": "road_class == MOTORWAY",
        "multiply_by": 0
      }
    ]
  },
  "profile": "car"
} 
```

"If you put this into a file route.json you can immediately try this:"

```
curl -XPOST -H "Content-Type: application/json" "http://localhost:8989/route" --data route.json
```

Reply:
Ok yes good idea

Reply:
> curl -XPOST -H "Content-Type: application/json" "http://localhost:8989/route" --data route.json

This requires a local server, shall we change localhost to the custom model demo server IP?

Reply:
I would like to get rid of these demo servers, when we get this feature on graphhopper.com ( hopefully soonish ;) )

And I think that the reader here will have the ability to set up a local routing server. For our docs at docs.graphhopper.com we probably have to copy these docs (?)

Reply:
Instead of copying the docs we could also just move them entirely? But for docs.graphhopper.com maybe we want to be specific about which encoded values exist etc. too. So far I thought custom-models.md is for client-side only users (like on GH maps) so they do not need to know about setting up a local server. Setting up a local server is described in profiles.md. But maybe we can just remove the comment about curl here because how to send a GH request is described elsewhere anyway.

Reply:
Yes, feel free to remove the curl command.

> Instead of copying the docs we could also just move them entirely? 

In general we cannot use the same docs here as we use in docs.graphhopper.com. I'm not sure if we would need a separate documentation for custom models and although introducing this separation causes lots of heading it also allows us to independently move forward here without "production" limitations.

Reply:
Ok so let's see when we actually do it and then probably copy it (or most parts of it)?

Reply:
:+1: